src/main/kotlin/pt/up/fe/ni/website/backend/service/PostService.kt

```diff
@@ -0,0 +1,33 @@
+package pt.up.fe.ni.website.backend.service
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import pt.up.fe.ni.website.backend.model.Post
+import pt.up.fe.ni.website.backend.repository.PostRepository
+
+@Service
+class PostService(private val repository: PostRepository) {
+    fun getAllPosts(): Collection<Post> = repository.findAll().toList()
+
+    fun getPost(postID : String): Post = repository.findByIdOrNull(postID.toLong()) ?: throw NoSuchElementException("Post Not Found")
+
+    fun createPost(post: Post) = repository.save(post)
+
+    fun updatePost(postID: String, newDescription: String?, newLink: String?) {
+        val targetPost = repository.findByIdOrNull(postID.toLong()) ?: throw NoSuchElementException("Post Not Found")
+        if (newDescription != null) {
+            targetPost.description = newDescription
+        };
+        if (newLink != null) {
+            targetPost.link = newLink
+        }
+
+        repository.save(targetPost)
+
+    }
+
+    fun deletePost(postID : String) {
+        val targetPost = repository.findByIdOrNull(postID.toLong()) ?: throw NoSuchElementException("Post Not Found")
+        repository.delete(targetPost)
+    }
+}
```

Comment:
This works nicely and I'm just leaving this comment for the record. We should probably find a nice way of validating the `postID` whenever it's used and throw `NoSuchElementException("Post Not Found")` there. That way, we could simplify this to something like this. We could think about it in #15 too
```suggestion
    fun deletePost(@ExistingPost postID : Long) {
        repository.deleteById(postID)
    }
```